### ISSUE TITLE
K.std results in NaN if input is zero vector

### ISSUE THREAD

**Issue by githubnemo** [fchollet/keras#2960]
Since `K.std(x)` is just `sqrt(var(x))` under the hood the gradient will be `inf` if the input vector `x` is 0.

This is problematic when using the `BatchNormalization` layer and having a model that sometimes produces zero vectors.

Wouldn't it be better to expose `K.var` in the backends and use that instead of `K.std` so that
the code can do `K.sqrt(K.var(x) + eps)` and does not have to add `eps` to `x`?

**Comment by fchollet** [comment]
That sounds reasonable. Do you want to make a PR?

**Comment by githubnemo** [comment]
Made a PR. I'm not sure if `K.std` is really needed anymore though.
